src/PostgREST/ApiRequest.hs

```diff
@@ -125,8 +126,11 @@ userApiRequest schema req reqBody
     case decodeContentType . fromMaybe "application/json" $ lookupHeader "content-type" of
       CTApplicationJSON ->
           either Left (\val -> case ensureUniform (pluralize val) of
-            Nothing -> Left "All object keys must match"
-            Just json -> Right json) (JSON.eitherDecode reqBody)
+              Nothing -> Left "All object keys must match"
+              Just json -> Right json)
+            (if BL.null reqBody && isTargetingProc
+              then Right emptyObject
+              else JSON.eitherDecode reqBody)
```

Comment:
actually `note` is already there since we use protolude
you should use that and not import anything else

https://github.com/sdiehl/protolude/blob/3994639b2b442f9ee2b0d3a197c5e2ecb579d3b7/src/Exceptions.hs#L30